### PR TITLE
feat(chat): add layout for shared auth logic, add marketplace route

### PR DESCRIPTION
**Pull request by Gimir**
**Description:**

- add layout for shared auth logic between pages
- add marketplace route
- add marketplace layout components: header, filterbar, card view (draft)

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

File: apps/chat/public/images/icons/home.svg
Comment (Alexander-Kezik) [inline apps/chat/public/images/icons/home.svg:1]: lets remove this and find similar icon in tabler icons

File: apps/chat/public/images/icons/arrow-left.svg
Comment (Alexander-Kezik) [inline apps/chat/public/images/icons/arrow-left.svg:1]: lets remove this and find similar icon in tabler icons

File: apps/chat/src/components/Marketplace/MarketplaceFilterbar.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  caption,
+  Icon,
+}: ActionButtonProps) => {
+  return (
+    <div className="flex px-2 py-1">
+      <button
+        onClick={onClick}
+        className="flex min-h-9 shrink-0 grow cursor-pointer select-none items-center gap-3 rounded px-4 py-2 transition-colors duration-200 hover:bg-accent-primary-alpha hover:disabled:bg-transparent"
+      >
+        {/* @ts-expect-error-next-line */}
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/MarketplaceFilterbar.tsx]: do we still need this comment?

File: apps/chat/src/components/Header/MarketplaceHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+import MoveLeftIcon from '../../../public/images/icons/move-left.svg';
+import MoveRightIcon from '../../../public/images/icons/move-right.svg';
+import { User } from './User/User';
+
+import { Feature } from '@epam/ai-dial-shared';
+import cssEscape from 'css.escape';
+
+const DEFAULT_HEADER_ICON_SIZE = 24;
+const OVERLAY_HEADER_ICON_SIZE = 18;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Header/MarketplaceHeader.tsx]: move these into constants file and use into both header components

File: apps/chat/src/components/Marketplace/MarketplaceFilterbar.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/* eslint-disable tailwindcss/no-custom-classname */
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/MarketplaceFilterbar.tsx]: Why do we need it?

File: apps/chat/src/components/Marketplace/MarketplaceFilterbar.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+import classnames from 'classnames';
+
+import { Translation } from '@/src/types/translation';
+
+import { useAppSelector } from '@/src/store/hooks';
+import { UISelectors } from '@/src/store/ui/ui.reducers';
+
+import ArrowIcon from '../../../public/images/icons/arrow-left.svg';
+import HomeIcon from '../../../public/images/icons/home.svg';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/MarketplaceFilterbar.tsx]: use tabler icons instead

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/515172)

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/515174)

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**